crates/uv-distribution/src/metadata.rs

```diff
@@ -28,6 +36,7 @@ pub struct Metadata {
     pub requires_dist: Vec<pypi_types::Requirement>,
     pub requires_python: Option<VersionSpecifiers>,
     pub provides_extras: Vec<ExtraName>,
+    pub dependency_groups: BTreeMap<ExtraName, Vec<pypi_types::Requirement>>,
```

Comment:
I find the terminology here really confusing -- that we have both extras and dependency groups -- but, what can we do? I'm honestly considering using `dev_dependencies` everywhere internally instead of dependency groups.